src/controlplane/controlplane_test.go

```diff
@@ -40,6 +40,17 @@ const (
 	clusterName    = "testClusterName"
 )
 
+// apiServer123DeprecationExclusion defines an specific exclusion introduced to support k8s 1.23.
+func apiServer123DeprecationExclusion(version testutil.Version) exclude.Func {
+	return func(group string, spec *definition.Spec, ent *integration.Entity) bool {
+		// apiserverStorageObjects replaces etcObjectCounts in k8s versions above 1.23
+		if testutil.IsBelow(version, testutil.Testdata123) {
+			return exclude.Metrics("apiserverStorageObjects")(group, spec, ent)
+		}
+		return exclude.Metrics("etcdObjectCounts")(group, spec, ent)
+	}
+}
```

Comment:
I don't like including this version-related check here, but I think this is something we need to test. I considered some alternatives:

- Defining the exclusions closer to version definition (something like `version.Exclusions()`): I discarded it because the exclusion only make sense for a particular control-plane tests.
- Excluding all optional metrics y the 'all control plane test' and create a new tests for this metric: I discarded it because it may sound simpler but it is more code and harder to read.

Please, let me know your thoughts.

Reply:
I get the idea but not why we are checking the version here, rather than outside: https://github.com/newrelic/nri-kubernetes/blob/3cd1b6b2ef50951fd129f62300e2d00eea6d7ace/src/controlplane/controlplane_test.go#L116

We could check the version in the test loop, and add the exclusion function for that test only, rather than checking the version inside the FxcludeFunc. WDYT?

Reply:
Well, it would be a bit different because we need to include a exclusion for every version (one when "version < 1.23", another when "version >= 1.23". So, including the version check in the loop would look like this:

```go
versionAsserter := asserter
if testutil.IsBelow(version, testutil.Testdata123) {
    versionAsserter = asserter.Exclude("apiserverStorageObjects")
else {
    versionAsserter = asserter.Exclude("etcdObjectCounts")
}
versionAsserter.On(i.Entitites).Assert(t)
```

If we added more version checks, I would not like to see the loop growing... So I prefer checking the version there.

Maybe the function name is confusing, maybe something like `apiserverObjectsExclusion` would be easier to read (the version is a parameter and now I see that one of its values should not be part of the function name).

Reply:
Updated here: 654bdea917bc89cd313853086eb9a08ac13ccbc2